notes-0407-composition-structure.tex

```diff
@@ -0,0 +1,303 @@
+\documentclass[11pt]{article}
+\usepackage{hdtt2020}
+\usepackage{newpxmath}
+\usepackage{newpxtext}
+\usepackage{multicol}
+%\usepackage{bussproof}
+
+\NewDocumentCommand{\circletype}{}{\mathtt{circle}}
+\NewDocumentCommand{\circleloop}{}{\mathtt{loop}}
+\NewDocumentCommand{\hcomp}{}{\mathtt{hcomp}}
+\NewDocumentCommand{\pathconcat}{mm}{\idconcat{#1}{#2}}
+\NewDocumentCommand{\pathrefl}{g}{\mathtt{refl}\IfValueT{#1}{\parens{#1}}}
+\NewDocumentCommand{\pathtype}{ggg}{\IfValueTF{#3}{\mathtt{Path}_{#1}\parens{#2; #3}}{\IfValueTF{#1}{\mathtt{Path}\parens{#1; #2}}{\mathtt{Path}}}}
+
+\title{CSCI 8980 Higher-Dimensional Type Theory\\ Lecture Notes}
+\author{Nathan Ringo, Norihiro Yamada}
+\date{April 7, 2020} % Also some Apr1 content for context
+
+\begin{document}
+\maketitle
+
+In this lecture note, we shall first introduce two operators (on terms) equipped on every type, called \emph{homogeneous composition} and \emph{coercion}, which together ensure that paths can be composed in every type. 
+We shall then see how these two operators are defined along the basic type constructions. 
+
+
+\section{Homogeneous Composition}
+\label{HomogeneousComposition}
+\emph{\bfseries Homogeneous composition} is concerned with an $n$-cube ($n \in \mathbb{N}$) with a missing term or \emph{\bfseries face}; we call such an $n$-cube a \emph{\bfseries box}, and the face of a box that is on the side opposite to the missing face the \emph{\bfseries cap}. 
+Homogenous composition is then what gives us the missing face of a box from the cap.
+
+%To make homogeneous composition useful in type theory, we extend it to boxes with multiple missing faces. 
+Let us remark that as long as the cap of a box is present we can apply homogeneous composition to the box \emph{even if the other faces are absent}.
+Also, to apply homogeneous composition, the faces of a box must share the \emph{same} type; this point explains the terminology \emph{homogeneous} composition. 
```

Comment:
The faces might still have different types. The composition is homogeneous because the type is not changing along the composition dimension, but it can still be changing in other dimensions.